master/internal/configpolicy/task_config_policy.go

```diff
@@ -28,7 +31,74 @@ type NTSCConfigPolicies struct {
 	Constraints     *model.Constraints   `json:"constraints"`
 }
 
-// PriorityAllowed returns true if the desired priority is within the limit set by task config policies.
+var (
+	errPriorityConstraintFailure = errors.New("submitted workload failed priority constraint")
+	errResourceConstraintFailure = errors.New("submitted workload failed a resource constraint")
+)
+
+// CheckNTSCConstraints returns true if the NTSC config passes constraint checks.
+func CheckNTSCConstraints(
+	ctx context.Context,
+	workspaceID int,
+	workloadConfig model.CommandConfig,
+	resourceManager rm.ResourceManager,
+) (bool, error) {
+	constraints, err := GetMergedConstraints(ctx, workspaceID, model.NTSCType)
+	if err != nil {
+		return false, err
+	}
+
+	// For each submitted constraint, check if the workload config is within allowed values.
+	// rm.SmallerValueIsHigherPriority only returns nil if priority is not implemented for that resource manager.
+	// In that case, there is no need to check if requested priority is within limits.
+	smallerHigher, err := resourceManager.SmallerValueIsHigherPriority()
+	if err == nil && constraints.PriorityLimit != nil && workloadConfig.Resources.Priority != nil {
+		if !priorityWithinLimit(*workloadConfig.Resources.Priority, *constraints.PriorityLimit, smallerHigher) {
+			return false, fmt.Errorf("requested priority [%d] exceeds limit set by admin [%d]: %w",
+				*constraints.PriorityLimit, *workloadConfig.Resources.Priority, errPriorityConstraintFailure)
+		}
+	}
+
+	if constraints.ResourceConstraints != nil && constraints.ResourceConstraints.MaxSlots != nil &&
+		workloadConfig.Resources.MaxSlots != nil {
+		if *constraints.ResourceConstraints.MaxSlots < *workloadConfig.Resources.MaxSlots {
+			return false, fmt.Errorf("requested resources.max_slots [%d] exceeds limit set by admin [%d]: %w",
+				*constraints.ResourceConstraints.MaxSlots, *workloadConfig.Resources.MaxSlots, errResourceConstraintFailure)
+		}
+	}
+
+	return true, nil
+}
+
+// GetMergedConstraints retrieves Workspace and Global constraints and returns a merged result.
+// workloadType is expected to be model.ExperimentType or model.NTSCType.
+func GetMergedConstraints(ctx context.Context, workspaceID int, workloadType string) (*model.Constraints, error) {
+	// Workspace-level constraints should be over-ridden by global contraints, if set.
+	var constraints model.Constraints
+	wkspConfigPolicies, err := GetTaskConfigPolicies(ctx, &workspaceID, workloadType)
+	if err != nil {
+		return nil, err
+	}
+	if wkspConfigPolicies.Constraints != nil {
+		if err = json.Unmarshal([]byte(*wkspConfigPolicies.Constraints), &constraints); err != nil {
+			return nil, fmt.Errorf("unable to merge workspace and global constraints: %w", err)
```

Comment:
nit: I kind of like the error message to be more specific to what went wrong in a given function if that makes sense. So if unmarshaling the constraints failed while trying to get the merged constraints, i'd rather have callers of `GetMergedConstraints` return the error "unable to merge workspace and global constraints", while this error message can say something like "unable to unmarshal workspace constraints"

Again, this is another preference-like nit

Reply:
Since this wraps the JSON unmarshaling error, both should show up.